### ISSUE TITLE
Cloud Build Trigger build option machine_type support for E2_MEDIUM

### ISSUE THREAD

**Issue by faddiekhan** [hashicorp/terraform-provider-google#15282]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

Within the cloudbuild.yaml, the REST API [1], as well as the Cloud Build UI, it is possible to specify `E2_MEDIUM` as `machine_type`. Since Google is changing the default machine type of cloud build from `E2_MEDIUM` to `E2-STANDARD` after **1st September**,  and `E2-STANDARD` is more expensive than `E2-MEDIUM` then it would be nice to have `E2_MEDIUM` machine_type as  one of the options via terraform as well.

Right now in terraform machine_type parameter only has one of these `["UNSPECIFIED", "N1_HIGHCPU_8", "N1_HIGHCPU_32", "E2_HIGHCPU_8", "E2_HIGHCPU_32"]` options as machine type to specify.

### New or Affected Resource(s)

* google_cloudbuild_trigger

### Potential Terraform Configuration

```tf
resource "google_cloudbuild_trigger" "build-trigger" {
  location = "global"

  trigger_template {
    branch_name = "main"
    repo_name   = "my-repo"
  }

  build {
    options {
         machine_type = "E2_MEDIUM" # Specify e2-medium machine type
         disk_size_gb = 100
   }
    step {
      name           = "ubuntu"
      script         = "echo hello"
    }

    step {
      name         = "python"
      script       = "#!/usr/bin/env python\nprint('hello from python')\n"
    }

    source {
      storage_source {
        bucket = "mybucket"
        object = "source_code.tar.gz"
      }
    }

    tags          = ["build", "newFeature"]
    substitutions = {
      _FOO = "bar"
      _BAZ = "qux"
    }
  }
}
```

### References

* [1] https://cloud.google.com/build/docs/api/reference/rest/v1/projects.builds#machinetype

* b/293782458

**Comment by DanielRieske** [comment]
I'd happily pick this issue up, I will add the PR shortly!

**Comment by rileykarson** [comment]
Note: We should consider making the field a string rather than an enum.

**Comment by DanielRieske** [comment]
@rileykarson As there are currently only five supported options I would still prefer keeping these strong typed, out of curiosity what the argument be for using strings here?

**Comment by rileykarson** [comment]
Frequency of updates is one of the considerations: https://google.aip.dev/126#when-to-use-enums

Machine types are _probably_ gonna update more than once a year, and unblocking new types when they do become available could be valuable.
